src/components/OfflineWithFeedback.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import {Pressable, View} from 'react-native';
+import PropTypes from 'prop-types';
+import compose from '../libs/compose';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import {withNetwork} from './OnyxProvider';
+import networkPropTypes from './networkPropTypes';
+import Text from './Text';
+import styles from '../styles/styles';
+import Tooltip from './Tooltip';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import * as StyleUtils from '../styles/StyleUtils';
+
+const propTypes = {
+    /** The type of action that's pending  */
+    pendingAction: PropTypes.oneOf(['add', 'update', 'delete']),
+
+    /** The error to display  */
+    error: PropTypes.string,
+
+    /** A function to run when the X button next to the error is clicked */
+    onClose: PropTypes.func.isRequired,
+
+    /** The content that needs offline feedback */
+    children: PropTypes.node.isRequired,
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    /** Additional styles to add after local styles. Applied to Pressable portion of button */
+    style: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.object),
+        PropTypes.object,
+    ]),
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    pendingAction: null,
+    error: null,
+    style: [],
+};
+
+class OfflineWithFeedback extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+        };
```

Comment:
Is this state going to be used? if you end up not using an state, you may want to turn this into a function component: https://github.com/Expensify/App/blob/c67d0d48c050ac0319ad66d38a37596f137844e7/contributingGuides/STYLE.md#stateless-components-vs-pure-components-vs-class-based-components-vs-render-props---when-to-use-what